x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable_factory.tsx

```diff
@@ -0,0 +1,213 @@
+        new BehaviorSubject(undefined);
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          ...timeRangeApi,
+          query$,
+          filters$,
+          // @ts-ignore
+          appliedTimeRange$,
+          jobIds,
+          swimlaneType,
+          viewBy,
+          fromPage,
+          perPage,
+          interval,
+          setInterval: (v) => interval.next(v),
+          updateUserInput: (update: AnomalySwimlaneEmbeddableUserInput) => {
+            jobIds.next(update.jobIds);
+            swimlaneType.next(update.swimlaneType);
+            viewBy.next(update.viewBy);
+            titlesApi.setPanelTitle(update.panelTitle);
+          },
+          updatePagination: (update: { perPage?: number; fromPage: number }) => {
+            fromPage.next(update.fromPage);
+            if (update.perPage) {
+              perPage.next(update.perPage);
+            }
+          },
+          dataViews: buildDataViewPublishingApi(
+            {
+              anomalyDetectorService: services[2].anomalyDetectorService,
+              dataViewsService: services[1].data.dataViews,
+            },
+            { jobIds },
+            subscriptions
+          ),
+          dataLoading: dataLoading$,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                jobIds: jobIds.value,
+                swimlaneType: swimlaneType.value,
+                viewBy: viewBy.value,
+                perPage: perPage.value,
+                timeRange: appliedTimeRange$.value,
+              },
+
+              references: [],
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          jobIds: [jobIds, jobIds.next, fastIsEqual],
+          swimlaneType: [swimlaneType, swimlaneType.next],
+          viewBy: [viewBy, viewBy.next],
+          perPage: [perPage, perPage.next],
+        }
+      );
+
+      const onError = () => {
+        dataLoading$.next(false);
+      };
+
+      const onLoading = () => {
+        dataLoading$.next(true);
+      };
+
+      const refresh$ = new BehaviorSubject<void>(undefined);
+
+      const onRenderComplete = () => {};
+
+      return {
+        api,
+        Component: () => {
+          const I18nContext = services[0].i18n.Context;
+          const theme = services[0].theme;
+
+          useEffect(function onUnmount() {
+            return () => {
+              subscriptions.unsubscribe();
+            };
+          }, []);
+
+          return (
+            <I18nContext>
+              <KibanaThemeProvider theme={theme}>
+                <KibanaContextProvider services={{ ...services[0] }}>
+                  <Suspense fallback={<EmbeddableLoading />}>
+                    <EmbeddableSwimLaneContainer
+                      api={api}
```

Comment:
Interesting strategy to put all state subjects on the API then send it into the component in this way. It's not a bad technique, I just haven't seen it done like this before!

Is the `EmbeddableSwimLaneContainer` ever used _outside_ of this embeddable?
Do you need items like the `perPage` or `fromPage` accessible outside of the Embeddable, like in actions or an editor etc?

Reply:
`EmbeddableSwimLaneContainer` is only for embeddable context. The purpose of this component is to fetch data needed for the swim lane component and retrieve required props from the embeddable input.